pandas/core/series.py

```diff
@@ -3575,19 +3581,70 @@ def sort_values(
         else:
             return result.__finalize__(self, method="sort_values")
 
-    @deprecate_nonkeyword_arguments(version=None, allowed_args=["self"])
+    # error: Signature of "sort_index" incompatible with supertype "NDFrame"
+    @overload  # type: ignore[override]
     def sort_index(
         self,
-        axis=0,
-        level=None,
-        ascending: bool | int | Sequence[bool | int] = True,
+        *,
+        axis: Literal[0] = ...,
```

Comment:
not sure about this. I've tried in the past to restrict the axis parameter to Literals and run into all sorts of issues.

This is compounded by the fact that the parameter is only really included for compatibility with DataFrame and can only be 0 for Series. (any maybe "index" although not documented?)

So the compatibility is to allow the user to maybe have an axis argument that would have a value of 0 when applied to a Series. Since they have the argument for compatibility it is most likely an int or str or maybe Literal[0,1,"index","columns"]. Either way, if it is not a Literal[0] they will get false postives.

in pandas._typing we currently define `Axis = Union[str, int]`. I see no reason to be more restrictive here for now, until Axis is defined as Literal[0,1,"index","columns"].

Also, type checking is to generally find TypeErrors whereas the wrong axis is a ValueError. (Given that's also the case wherever Literal is used, I appreciate that this is not a strong argument and it's a grey area as to when Literals are useful)

all things considered, I think the compatibility with DataFrame argument is the strongest and therefore use the same type annotation for Series and DataFrame?

Reply:
Theoretically, we should have the same interface as NDFrame but Series errors in some cases. I think it would be more explicit to signal the stricter type through the type annotations. Happy either way.

Reply:
> I think it would be more explicit to signal the stricter type through the type annotations.

I agree that makes total sense from a typing perspective but at the same time seems to be at odds with the raison d'être of the axis argument for Series.

Reply:
done